mkt/extensions/validation.py

```diff
@@ -32,10 +33,18 @@ class ExtensionValidator(object):
         'DESCRIPTION_TOO_LONG': _(
             u'The `description` property cannot be '
             u'longer than 132 characters.'),
-        'ICONS_DO_NOT_EXIST': _(u'Not all specified icons exist.'),
+        'ICON_INCORRECT_DIMENSIONS': _(
+            u'The icon file `%s` dimensions are different from the size '
+            u'in the manifest'),
+        'ICON_DOES_NOT_EXIST': _(
+            u'The icon file `%s` is referenced in the manifest but does not'
+            u' exist in the ZIP file.'),
         'ICONS_NO_128': _(
             u'If defining `icons`, you must include a 128x128 variant.'),
-        'ICONS_INVALID_FORMAT': _(u'Only PNG icons are permitted.'),
+        'ICON_NOT_A_VALID_IMAGE_OR_PNG': _(
+            u'The icon file `%s` is not a valid image or not in PNG format.'),
```

Comment:
The icon file `%s` is not a valid PNG.

This reads better and is equally accurate (invalid images are also not PNGs).